pkg/providers/amifamily/resolver.go

```diff
@@ -130,21 +131,32 @@ func (r Resolver) Resolve(ctx context.Context, nodeClass *v1beta1.EC2NodeClass,
 	}
 	var resolvedTemplates []*LaunchTemplate
 	for amiID, instanceTypes := range mappedAMIs {
-		maxPodsToInstanceTypes := lo.GroupBy(instanceTypes, func(instanceType *cloudprovider.InstanceType) int {
-			return int(instanceType.Capacity.Pods().Value())
+		type launchTemplateParams struct {
+			efaCount int
+			maxPods  int
+		}
+		paramsToInstanceTypes := lo.GroupBy(instanceTypes, func(instanceType *cloudprovider.InstanceType) launchTemplateParams {
```

Comment:
How much does this increase the number of launch templates that we create in the worst-case? Would be good to do some analysis on how much this is going to increase our TPS toward `CreateLaunchTempmlate`

Reply:
I'd suspect that we're roughly doubling our launch template count here

Reply:
Threw together a little script to test this and here are the results:
```
# With EFAs
EFA Count: 0, Total Max Pod Values: 21, Max Pod Values: [44 437 59 35 10 78 118 394 17 20 234 737 29 18 38 89 58 4 8 12 11]
EFA Count: 1, Total Max Pod Values: 6, Max Pod Values: [737 198 234 58 321 345]
EFA Count: 2, Total Max Pod Values: 1, Max Pod Values: [345]
EFA Count: 4, Total Max Pod Values: 1, Max Pod Values: [737]
EFA Count: 8, Total Max Pod Values: 1, Max Pod Values: [247]
EFA Count: 16, Total Max Pod Values: 1, Max Pod Values: [247]
EFA Count: 32, Total Max Pod Values: 1, Max Pod Values: [100]
Total Possible Launch Templates: 32
```
```
# Without EFAs
Total Max Pod Values: 26, Max Pod Values: [58 89 737 29 198 118 8 234 12 4 78 18 247 17 10 59 35 11 321 20 100 394 345 44 437 38]
Total Possible Launch Templates: 26
```
We already discussed this offline but the increase of 6 is pretty negligible. We've got a burst limit of 200 requests for create calls. This is based off instances available in `us-east-1` but if anything there will be less instance types in other regions (generally).